lib/Sema/CSDiag.cpp

```diff
@@ -2430,6 +2430,12 @@ bool FailureDiagnosis::diagnoseGeneralMemberFailure(Constraint *constraint) {
     }
   }
 
+  // If base type has unresolved generic parameters, such might mean
+  // that it's initializer with erroneous argument, otherwise this would
+  // be a simple ambigious archetype case, neither can be diagnosed here.
+  if (baseTy->hasTypeParameter())
```

Comment:
It's kind of weird to me, but this `<<unresolved>>.Iterator.Element is actually considered a type parameter so maybe I should just check for both - `if (baseTy->hasTypeParameter() && baseTy->hasUnresolved())` ?